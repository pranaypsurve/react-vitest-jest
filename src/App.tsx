import { useState } from "react";
import reactLogo from "./assets/react.svg";
import viteLogo from "/vite.svg";
import "./App.css";

function App() {
  const [count, setCount] = useState(0);
  const [username, setUsername] = useState("");
  const [text, setText] = useState("");

  return (
    <>
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react pranay" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      {/* <WebRTCComponent /> */}
      <button title="count is" onClick={() => setCount((count) => count + 1)}>
        count is {count}
      </button>
      <div>
        <button onClick={() => setText("hello")}>Click</button>
        <h2>{text}</h2>
        <input
          type="text"
          name="username"
          placeholder="Name"
          onChange={(e) => setUsername(e.target.value)}
          value={username}
          readOnly
        />
      </div>
    </>
  );
}

export default App;
