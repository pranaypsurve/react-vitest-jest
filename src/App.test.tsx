import { fireEvent, render, screen } from "@testing-library/react";
import { describe, it } from "vitest";
import App from "./App";

describe("snapshot", () => {
  it("app snapshot", () => {
    const conatiner = render(<App />);
    expect(conatiner).toMatchSnapshot();
  });
});

describe("#App Test Cases", () => {
  it("Check Title Text", () => {
    render(<App />);
    const titleText = screen.getByText("Vite + React");
    expect(titleText).toBeInTheDocument();
  });

  it("Check count title", () => {
    render(<App />);
    const text2 = screen.getByTitle(/count is/i);
    expect(text2).toBeInTheDocument();
  });

  it("test input field login", () => {
    render(<App />);
    const inputBox: HTMLElement[] = screen.getAllByRole("textbox");
    inputBox.forEach((item) => {
      expect(item).toBeInTheDocument();
      expect(item).toHaveAttribute("type");
      expect(item).toHaveAttribute("name");
      expect(item).toHaveAttribute("placeholder");
    });
    fireEvent.change(inputBox[0], { target: { value: "surve" } });
    expect(inputBox[0].getAttribute("value")).toBe("surve");
  });
  describe("inner button text", () => {
    it("button text demo", () => {
      render(<App />);
      const btn = screen.getAllByRole("button")[1];
      fireEvent.click(btn);
      expect(screen.getByText("hello")).toBeInTheDocument();
    });
  });
});
