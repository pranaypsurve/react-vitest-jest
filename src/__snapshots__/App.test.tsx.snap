// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot > app snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          href="https://vitejs.dev"
          target="_blank"
        >
          <img
            alt="Vite logo"
            class="logo"
            src="/vite.svg"
          />
        </a>
        <a
          href="https://react.dev"
          target="_blank"
        >
          <img
            alt="React logo"
            class="logo react pranay"
            src="/src/assets/react.svg"
          />
        </a>
      </div>
      <h1>
        Vite + React
      </h1>
      <button
        title="count is"
      >
        count is 
        0
      </button>
      <div>
        <button>
          Click
        </button>
        <h2 />
        <input
          name="username"
          placeholder="Name"
          readonly=""
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        href="https://vitejs.dev"
        target="_blank"
      >
        <img
          alt="Vite logo"
          class="logo"
          src="/vite.svg"
        />
      </a>
      <a
        href="https://react.dev"
        target="_blank"
      >
        <img
          alt="React logo"
          class="logo react pranay"
          src="/src/assets/react.svg"
        />
      </a>
    </div>
    <h1>
      Vite + React
    </h1>
    <button
      title="count is"
    >
      count is 
      0
    </button>
    <div>
      <button>
        Click
      </button>
      <h2 />
      <input
        name="username"
        placeholder="Name"
        readonly=""
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
